pennylane/fourier/spectrum.py

```diff
@@ -0,0 +1,261 @@
+    r"""Join two sets of frequencies that belong to the same input scalar.
+
+    Since :math:`\exp(i a x)\exp(i b x) = \exp(i (a+b) x)`, frequency sets of two gates
+    encoding the same :math:`x` are joined by computing the set of sums of their elements.
+
+    Args:
+        spec1 (list[float]): first spectrum
+        spec2 (list[float]): second spectrum
+    Returns:
+        list[float]: joined spectrum
+    """
+    sums = [s1 + s2 for s1 in spec1 for s2 in spec2]
+    return sorted(list(set(sums)))
+
+
+def spectrum(qnode, encoding_gates=None):
+    r"""Compute the frequency spectrum of the Fourier representation of simple quantum circuits.
+
+    The circuit must only use single-parameter gates of the form :math:`e^{-i x_j G}` as
+    input-encoding gates, which allows the computation of the spectrum by inspecting the gates'
+    generators :math:`G`.
+
+    Gates are marked as input-encoding gates in the quantum function by giving them an ``id``.
+    If two gates have the same ``id``, they are considered
+    to be used to encode the same input :math:`x_j`. The `encoding_gates` argument can be used
+    to indicate that only gates with a specific ``id`` should be interpreted as input-encoding gates.
+
+    Args:
+        qnode (pennylane.QNode): a quantum node representing a circuit in which
+            input-encoding gates are marked by their ``id`` attribute
+        encoding_gates (list[str]): list of input-encoding gate ``id`` strings
+            for which to compute the frequency spectra
+
+    Returns:
+        (dict[str, list[float]]): Dictionary with the input-encoding gate ``id`` as keys and
+        their frequency spectra as values.
+
+
+    **Details**
+
+    A circuit that returns an expectation value which depends on
+    :math:`N` scalar inputs :math:`x_j` can be interpreted as a function
+    :math:`f: \mathbb{R}^N \rightarrow \mathbb{R}`. This function can always be
+    expressed by a Fourier-type sum
+
+    .. math::
+
+        \sum \limits_{\omega_1\in \Omega_1} \dots \sum \limits_{\omega_N \in \Omega_N}
+        c_{\omega_1,\dots, \omega_N} e^{-i x_1 \omega_1} \dots e^{-i x_N \omega_N}
+
+    over the *frequency spectra* :math:`\Omega_j \subseteq \mathbb{R},`
+    :math:`j=1,\dots,N`. Each spectrum has the property that
+    :math:`0 \in \Omega_j`, and the spectrum is
+    symmetric (for every :math:`\omega \in \Omega_j` we have that :math:`-\omega \in
+    \Omega_j`). If all frequencies are integer-valued, the Fourier sum becomes a
+    *Fourier series*.
+
+    As shown in `Vidal and Theis (2019)
+    <https://arxiv.org/abs/1901.11434>`_ and `Schuld, Sweke and Meyer (2020)
+    <https://arxiv.org/abs/2008.08605>`_, if an input :math:`x_j, j = 1 \dots N`
+    only enters into single-parameter gates of the form :math:`e^{-i x_j G}`, the
+    frequency spectrum :math:`\Omega_j` is fully determined by the eigenvalues
+    of the generators :math:`G`. In many situations, the spectra are limited
```

Comment:
```suggestion
    only enters into single-parameter gates of the form :math:`e^{-i x_j G}` (where :math:`G` is a Hermitian generator),
    the frequency spectrum :math:`\Omega_j` is fully determined by the eigenvalues
    of :math:`G`. In many situations, the spectra are limited
```
In principle the G can be different, e.g., you can encode x_j into an RX or an RY. Do you think it's worth adding a subscript to the G?